tt_metal/impl/dispatch/command_queue.cpp

```diff
@@ -211,17 +240,17 @@ void EnqueueWriteInterleavedBufferCommand::add_buffer_data(HugepageDeviceCommand
         for (uint32_t sysmem_address_offset = 0; sysmem_address_offset < data_size_bytes;
              sysmem_address_offset += this->padded_page_size) {
             uint32_t page_size_to_copy = this->padded_page_size;
-            if (src_address_offset + this->padded_page_size > buffer.page_size()) {
+            if (src_address_offset + this->padded_page_size > this->buffer.page_size()) {
                 // last partial page being copied from unpadded src buffer
                 page_size_to_copy -= padding;
             }
             command_sequence.add_data((char*)this->src + src_address_offset, page_size_to_copy, this->padded_page_size);
             src_address_offset += page_size_to_copy;
         }
     } else {
-        uint32_t unpadded_src_offset =
-            (((buffer_addr_offset / this->padded_page_size) * num_banks) + this->dst_page_index) *
-            this->buffer.page_size();
+        uint32_t unpadded_src_offset = (((buffer_addr_offset / this->padded_page_size) * num_banks) +
+                                        (this->dst_page_index - this->orig_dst_page_index)) *
+                                       this->buffer.page_size();
```

Comment:
Instead of passing in `orig_dst_page_index`, can we just have the enqueue_write_buffer fn compute/pass in `unpadded_src_offset` itself? I think that makes things clearer